OptimizelySDK/ProjectConfig.cs

```diff
@@ -532,5 +541,18 @@ public Rollout GetRolloutFromId(string rolloutId)
             ErrorHandler.HandleError(new Exceptions.InvalidRolloutException("Provided rollout is not in datafile."));
             return new Rollout();
         }
+
+        /// <summary>
+        /// Get experiment from variation ID.
+        /// </summary>
+        /// <param name="variationId">Variation ID</param>
+        /// <returns>Experiment Entity corresponding to the variation ID or a dummy entity if ID is invalid</returns>
+        public Experiment GetExperimentForVariationId(string variationId)
+        {
+            if (_VariationIdToExperimentMap.ContainsKey(variationId))
```

Comment:
At least log a warning when no variation id is not found in the map

Reply:
Done